### ISSUE TITLE
JetFormBuilder in JetEngine listing shows success message for all forms

### ISSUE THREAD

**Issue by rtpHarry** [CrocoBlock/suggestions#5678]
**Describe the bug**
If you submit a JetFormBuilder form that is embedded inside a JetEngine Listing then it shows the success message for all of the forms, despite only one being actually submitted.

**To Reproduce**
Steps to reproduce the behaviour:

1. Make a custom post type, eg "volunteer events"
2. Make a JetFormBuilder form, eg "volunteer" (I'm attempting to write my own hook for this, which is being successfully called, so my form is set to Call hook in the actions)
3. Submit the form for one of the embedded forms
4. See error in screenshot below, it is showing the success message on all the forms, not just the one that was submitted.

**Expected behavior**
Only the submitted form should show its success message.

**Screenshots**
<img width="1183" alt="image" src="https://user-images.githubusercontent.com/1038062/180613169-6712bba9-ad5a-40b0-8a58-15f777d8a7b2.png">

**Plugin Versions**
 - JetFormBuilder v2.1.2
 - JetEngine v3.0.1

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Chrome Version 103.0.5060.114 (Official Build) (x86_64)

**Comment by rtpHarry** [comment]
It seems that somebody else has bumped into a similar issue like this back in 2020: #1966

**Comment by rtpHarry** [comment]
I actually solved a similar issue with the Elementor Image Carousel not having a unique id when used in a listing, but from playing around with a customised version of the code I cannot see any id values that can be modified this time around.

```php
<?php
/*
Plugin Name: SOKA Elementor Image Carousel Gallery ID
Plugin URI: https://sokastudio.co.uk/
Text Domain: soka-elementor-gallery-id
Description: Control the gallery id to separate out lightbox galleries in JetEngine listings
Version: 1.2.0
Author: Matthew Harris, SOKA Studio
Author URI: https://sokastudio.co.uk/
License: GPLv2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html

Copyright 2021 Matthew Harris

this program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License, version 2, as
published by the Free Software Foundation.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */

if ( ! defined( 'WPINC' ) ) {
	die;
}

/**
 * Customise the output html of image-carousels
 * 
 * @since 1.0.0
 * @see https://code.elementor.com/php-hooks/
 */
add_action( 'elementor/widget/render_content', function( $content, $widget ) {
    if ( 'image-carousel' === $widget->get_name() ) {
        $settings = $widget->get_settings();
   
        if ( ! empty( $settings['random_lightbox_gallery_id'] ) ) {
            $content = soka_modify_gallery_id($content);
        }
    }
    
    return $content;
}, 10, 2 );


/**
 * Inject a random gallery id prefix 
 * 
 * @since 1.0.0
 */
function soka_modify_gallery_id($content) {
    $key = 'data-elementor-lightbox-slideshow="';
    $random_id = apply_filters('soka/elementor/image_carousel/gallery_id/random_id', rand(1000, 9999) . '-' . rand(1000,9999));
    $key_prefixed = $key . $random_id . '-';

    $content = str_replace($key, $key_prefixed, $content);

    return $content;
}

/**
 * Add a control to toggle it on and off
 * 
 * @since 1.1.0
 */
add_action( 'elementor/element/image-carousel/section_image_carousel/before_section_end', function( $element, $args ) {
	/** @var \Elementor\Element_Base $element */
	$element->add_control(
        'random_lightbox_gallery_id',
        [
            'label' => __( 'Random Lightbox Gallery ID', 'soka-elementor-gallery-id' ),
            'description' => __('Prefix lightbox with a random gallery id', 'soka-elementor-gallery-id'),
            'type' => \Elementor\Controls_Manager::SWITCHER,
            'label_on' => __( 'Yes', 'soka-elementor-gallery-id' ),
            'label_off' => __( 'No', 'soka-elementor-gallery-id' ),
            'return_value' => 'yes',
            'default' => '',
            'condition' => [
                'open_lightbox!' => 'no',
            ],
        ]
    );
}, 10, 2 );
```

**Comment by rtpHarry** [comment]
It seems like JetPopup has solved this problem somehow, as it manages to handle having unique popups used inside of a JetEngine listing.

My guess is that the solution to this issue is going to be something to do with detecting if its inside a listing and then putting some marker on the form, and then the form will need to check if its the unique instance before sending back its validation/message results.

**Comment by rtpHarry** [comment]
Actually, there is a field in each of my forms `__queried_post_id` which is unique for each form, as its pointing at the parent post id of that listing.

I've just renamed this to imply that its a JetEngine issue, but it might be something that can be handled by JetFormBuilder if its aware of its listings post_id.

**Comment by VictorPietro** [comment]
Hey @rtpHarry it's been a while, but I've been struggling with this. I have multiple forms on the same page and I can't seem to find a solution to display only a submit message on the respective form - it just shows on all forms. Have you find an alternative?

**Comment by rtpHarry** [comment]
No, I think I spoke to the support team about it, but they just said that was expected behaviour, and any progress would be in this github issue.

The way I solved it was that I had to put the form inside a JetPopup, so that there wasn't multiple forms directly on the page.

But that causes its own problems, like the spam protection has edge cases then, I think. And you have to make the user manually close the popup after they submit, or the success message is lost.

I haven't built anything with these two technologies for a while so its a bit hazy in my mind. 

I think the other way that I solved it was to redirect to a success page.

Neither are great solutions.

I've had tons of trouble with JetFormBuilder every time I've tried to use it to be honest. It's far from being a competitor to the big names. For example password reset form is a total mindbending experience to manage their special states. When I used the multi-page plugin it cause random validation failures at the end of a long process where it was just stuck and wouldn't submit. When I used the user management addon there was lots of places where the error states didn't have their error messages properly set up which exposed cryptic errors to the user, and had hooks missing.

In their defence, they have actually implemented a number of my suggestions and were actively developing it, but I feel like I'm just testing their product for free and investing all of the time to try to map things out, and its a total gamble if my efforts are even going to make any progress, like this ticket.

**Comment by VictorPietro** [comment]
@rtpHarry hey I really appreciate your response and also your inputs on this.

It is a pity, indeed, because their plugin does have potential. I've been using JFB for a while and the ability to use hooks have been great on the project I'm currently working on. On the other hand, details such as this one are definitely frustrating. And perhaps I may be "content" with JFB because of my lack of experience with other advanced form plugins.

To get around it this issue, I've done similarly as you - using a JetPopup to display the resulting form submission message. It's not perfect, but it should be enough for now - although it apparently doesn't work with forms inside a JetPopup.

I've also had my share of bugs with JFB and the random validation errors are occasional. Fortunately in my case these forms are inside an area for logged-in users, so I could just disable the validations to get rid of this bug.

I do enjoy a lot Crocoblock team and their constant development progresses, but I also understand and share some of your frustrations.
